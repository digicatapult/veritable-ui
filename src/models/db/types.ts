import { Knex } from 'knex'
import { z } from 'zod'
import { carbonEmbodimentRequestData, carbonEmbodimentResponseData } from '../drpc.js'

export const tablesList = [
  'connection',
  'connection_invite',
  'query',
  'query_rpc',
  'settings',
  'organisation_registries',
] as const

const insertConnection = z.object({
  company_name: z.string(),
  company_number: z.string(),
  status: z.union([
    z.literal('pending'),
    z.literal('unverified'),
    z.literal('verified_them'),
    z.literal('verified_us'),
    z.literal('verified_both'),
    z.literal('disconnected'),
  ]),
  agent_connection_id: z.union([z.string(), z.null()]),
  pin_attempt_count: z.number().int().gte(0).lte(255),
  pin_tries_remaining_count: z.number().int().gte(0).lte(255).nullable(),
  registry_country_code: z.string(),
})

const insertOrganisationRegistries = z.object({
  country_code: z.string(),
  registry_name: z.string(),
  registry_key: z.string(),
  url: z.string(),
  api_key: z.string(),
})
const insertConnectionInvite = z.object({
  connection_id: z.string(),
  oob_invite_id: z.string(),
  pin_hash: z.string(),
  expires_at: z.date(),
  validity: z.union([z.literal('valid'), z.literal('expired'), z.literal('too_many_attempts'), z.literal('used')]),
})

const insertQueryShared = z.object({
  connection_id: z.string(),
  parent_id: z.string().nullable().optional(),
  type: z.literal('total_carbon_embodiment'),
  status: z.enum(['resolved', 'pending_your_input', 'pending_their_input', 'errored', 'forwarded']),
  response_id: z.string().nullable(),
  role: z.enum(['requester', 'responder']),
  expires_at: z.date(),
})
const totalCarbonEmbodimentQuery = z.object({
  type: z.literal('total_carbon_embodiment'),
  details: carbonEmbodimentRequestData,
  response: carbonEmbodimentResponseData.nullable(),
})
const insertQuery = z.discriminatedUnion('type', [totalCarbonEmbodimentQuery.merge(insertQueryShared)])

const defaultFields = z.object({
  id: z.string(),
  created_at: z.date(),
  updated_at: z.date(),
})

const insertQueryRpc = z.object({
  query_id: z.string(),
  agent_rpc_id: z.string(),
  role: z.union([z.literal('client'), z.literal('server')]),
  method: z.union([z.literal('submit_query_request'), z.literal('submit_query_response')]),
  result: z.union([z.record(z.any()), z.null()]).optional(),
  error: z.union([z.record(z.any()), z.null()]).optional(),
})
const insertSettings = z.object({
  setting_key: z.string(),
  setting_value: z.string(),
})

const Zod = {
  connection: {
    insert: insertConnection,
    get: insertConnection.merge(defaultFields),
  },
  connection_invite: {
    insert: insertConnectionInvite,
    get: insertConnectionInvite.merge(defaultFields),
  },
  query: {
    insert: insertQuery,
    get: z.discriminatedUnion('type', [totalCarbonEmbodimentQuery.merge(insertQueryShared).merge(defaultFields)]),
  },
  query_rpc: {
    insert: insertQueryRpc,
    get: insertQueryRpc.merge(defaultFields),
  },
  settings: {
    insert: insertSettings,
    get: insertSettings.extend({
      id: z.string(),
      created_at: z.date(),
      updated_at: z.date(),
    }),
  },
  organisation_registries: {
    insert: insertOrganisationRegistries,
    get: insertOrganisationRegistries.extend({
      id: z.string(),
    }),
  },
}

export type InsertConnection = z.infer<typeof Zod.connection.insert>
export type ConnectionRow = z.infer<typeof Zod.connection.get>
export type QueryRow = z.infer<typeof Zod.query.get>
export type SettingsRow = z.infer<typeof Zod.settings.get>
export type OrganisationRegistriesRow = z.infer<typeof Zod.organisation_registries.get>
export type TABLES_TUPLE = typeof tablesList
export type TABLE = TABLES_TUPLE[number]
export type Models = {
  [key in TABLE]: {
    get: z.infer<(typeof Zod)[key]['get']>
    insert: z.infer<(typeof Zod)[key]['insert']>
  }
}

export type ColumnsByType<M extends TABLE, T> = {
  [K in keyof Models[M]['get']]-?: Models[M]['get'][K] extends T ? K : never
}[keyof Models[M]['get']]

type WhereComparison<M extends TABLE> = {
  [key in keyof Models[M]['get']]: [
    Extract<key, string>,
    '=' | '>' | '>=' | '<' | '<=' | '<>' | 'LIKE' | 'ILIKE',
    Extract<Models[M]['get'][key], Knex.Value>,
  ]
}
export type WhereMatch<M extends TABLE> = {
  [key in keyof Models[M]['get']]?: Models[M]['get'][key]
}

export type Where<M extends TABLE> = WhereMatch<M> | (WhereMatch<M> | WhereComparison<M>[keyof Models[M]['get']])[]
export type Order<M extends TABLE> = [keyof Models[M]['get'], 'asc' | 'desc'][]
export type Update<M extends TABLE> = Partial<Models[M]['get']>

export type IDatabase = {
  [key in TABLE]: () => Knex.QueryBuilder
}

export default Zod
