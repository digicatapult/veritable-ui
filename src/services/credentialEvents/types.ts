import { Credential, CredentialFormatData } from '../../models/veritableCloudagent/internal.js'

export interface CredentialEventHandlerBase {
  handleProposalReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleOfferReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleRequestReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleCredentialReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleDone(credential: Credential, formatData: CredentialFormatData): Promise<void>
}

export interface CredentialEventHandler<EventName extends string, EventVersion extends string>
  extends CredentialEventHandlerBase {
  get schemaName(): EventName
  get schemaVersion(): EventVersion
}
