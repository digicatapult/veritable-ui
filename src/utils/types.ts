export type DiscriminateUnion<T, K extends keyof T, V extends T[K]> = T extends Record<K, V> ? T : never

export type MapDiscriminatedUnion<T extends Record<K, string>, K extends keyof T> = {
  [V in T[K]]: DiscriminateUnion<T, K, V>
}
