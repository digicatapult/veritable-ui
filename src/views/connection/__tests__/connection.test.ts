import { expect } from 'chai'
import { describe, it } from 'mocha'

import { ConnectionRow } from '../../../models/db/types.js'
import ConnectionTemplates from '../connection.js'

describe('ConnectionTemplates', () => {
  describe('listPage', () => {
    it('should render with no connections', async () => {
      const templates = new ConnectionTemplates()
      const rendered = await templates.listPage([])
      expect(rendered).to.matchSnapshot()
    })

    it('should render with single connection', async () => {
      const templates = new ConnectionTemplates()
      const rendered = await templates.listPage([
        {
          company_name: 'Company A',
          status: 'disconnected',
        },
      ] as ConnectionRow[])
      expect(rendered).to.matchSnapshot()
    })

    it('should escape html in name', async () => {
      const templates = new ConnectionTemplates()
      const rendered = await templates.listPage([
        {
          company_name: '<div>I own you</div>',
          status: 'verified_both',
        },
      ] as ConnectionRow[])
      expect(rendered).to.matchSnapshot()
    })

    it('should render multiple with each status', async () => {
      const templates = new ConnectionTemplates()
      const rendered = await templates.listPage([
        {
          company_name: 'Company A',
          status: 'disconnected',
        },
        {
          company_name: 'Company B',
          status: 'pending',
        },
        {
          company_name: 'Company C',
          status: 'unverified',
        },
        {
          company_name: 'Company D',
          status: 'verified_both',
        },
        {
          company_name: 'Company E',
          status: 'verified_them',
        },
        {
          company_name: 'Company F',
          status: 'verified_us',
        },
      ] as ConnectionRow[])
      expect(rendered).to.matchSnapshot()
    })
  })
})
