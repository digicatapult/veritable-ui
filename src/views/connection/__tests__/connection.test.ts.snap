// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTemplates listPage should escape html in name 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - Connections</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Connections</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connection Management\\" href=\\"#\\">Connection Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/connection\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"list-page-header\\"><span>Connection Management</span><a class=\\"button \\" href=\\"connection/new\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Invite New Connection</span></a><a class=\\"button \\" href=\\"connection/new?fromInvite=true\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Add from Invitation</span></a></div><div class=\\"list-page\\"><div class=\\"list-nav\\"><span>Connections</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search by Company Name\\" hx-get=\\"/connection\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Status</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>&lt;div>I own you&lt;/div></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Connected</div></td><td><a class=\\"button \\" href=\\"/queries/new?connectionId=someId\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Send Query</span></a></td></tr></tbody></table></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render multiple with each status 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - Connections</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Connections</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connection Management\\" href=\\"#\\">Connection Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/connection\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"list-page-header\\"><span>Connection Management</span><a class=\\"button \\" href=\\"connection/new\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Invite New Connection</span></a><a class=\\"button \\" href=\\"connection/new?fromInvite=true\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Add from Invitation</span></a></div><div class=\\"list-page\\"><div class=\\"list-nav\\"><span>Connections</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search by Company Name\\" hx-get=\\"/connection\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Status</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>Company A</td><td><div class=\\"list-item-status\\" data-status=\\"disabled\\">Disconnected</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Disconnected</span></a></td></tr><tr><td>Company B</td><td><div class=\\"list-item-status\\" data-status=\\"disabled\\">Invite Sent</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Waiting for Response</span></a></td></tr><tr><td>Company C</td><td><div class=\\"list-item-status\\" data-status=\\"warning\\">Verification Code Required</div></td><td><a class=\\"button \\" href=\\"/connection/someId/pin-submission\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Complete Verification</span></a></td></tr><tr><td>Company D</td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Connected</div></td><td><a class=\\"button \\" href=\\"/queries/new?connectionId=someId\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Send Query</span></a></td></tr><tr><td>Company E</td><td><div class=\\"list-item-status\\" data-status=\\"warning\\">Verification Code Required</div></td><td><a class=\\"button \\" href=\\"/connection/someId/pin-submission\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Complete Verification</span></a></td></tr><tr><td>Company F</td><td><div class=\\"list-item-status\\" data-status=\\"disabled\\">Waiting for Response</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Waiting for Response</span></a></td></tr></tbody></table></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with no connections 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - Connections</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Connections</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connection Management\\" href=\\"#\\">Connection Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/connection\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"list-page-header\\"><span>Connection Management</span><a class=\\"button \\" href=\\"connection/new\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Invite New Connection</span></a><a class=\\"button \\" href=\\"connection/new?fromInvite=true\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Add from Invitation</span></a></div><div class=\\"list-page\\"><div class=\\"list-nav\\"><span>Connections</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search by Company Name\\" hx-get=\\"/connection\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Status</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>No Connections for that search query. Try again or add a new connection</td></tr></tbody></table></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with single connection 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - Connections</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Connections</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connection Management\\" href=\\"#\\">Connection Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/connection\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#connection-search-input\\"><div class=\\"list-page-header\\"><span>Connection Management</span><a class=\\"button \\" href=\\"connection/new\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Invite New Connection</span></a><a class=\\"button \\" href=\\"connection/new?fromInvite=true\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>Add from Invitation</span></a></div><div class=\\"list-page\\"><div class=\\"list-nav\\"><span>Connections</span><input id=\\"connection-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search by Company Name\\" hx-get=\\"/connection\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Status</span><a class=\\"list-filter-icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>Company A</td><td><div class=\\"list-item-status\\" data-status=\\"disabled\\">Disconnected</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/arrow-right-circle.svg&#34;);\\"></div><span>Disconnected</span></a></td></tr></tbody></table></div></div></main></body></html>"`;
